### PR TITLE
[AlexNet] Refactored testing code to use pytest

### PR DESCRIPTION
**Pull request by tian0106**

### CODE REVIEW

File: models/cnn/AlexNet/test_AlexNet.py [hunk 1]
```diff
@@ -29,7 +30,148 @@ def test_overfit(model):
 
 
 if __name__ == "__main__":
```
Comment (danielmao2019) [inline models/cnn/AlexNet/test_AlexNet.py:61]: Don't use the main function. We are using pytest so just decorate the functions with @pytest.xxx.xxx. pytest would recognize the files of pattern test_*.py and run them. In your local, run `pytest models/cnn/AlexNet/test_AlexNet.py`.

File: models/cnn/AlexNet/test_AlexNet.py [hunk 2]
```diff
@@ -29,7 +30,148 @@ def test_overfit(model):
...
+    # num_classes = 10
+    #case 1 
     model = AlexNet(num_classes=10)
     model = model.build(input_shape=(128, 128, 3))
     test_forward_pass(model)
     test_overfit(model)
+    #case 2
```
Comment (danielmao2019) [inline models/cnn/AlexNet/test_AlexNet.py]: For running multiple test cases, use @pytest.mark.parametrize as I did for test_SegNet.py.

File: models/cnn/AlexNet/test_AlexNet.py [hunk 3]
```diff
@@ -29,7 +30,148 @@ def test_overfit(model):
...
+    model = AlexNet(num_classes=4)
+    model = model.build(input_shape=(225, 225, 3))
+    test_forward_pass(model)
+    test_overfit(model)
+    #case 3
+    model = AlexNet(num_classes=4)
+    model = model.build(input_shape=(128, 128, 1))
+    test_forward_pass(model)
+    test_overfit(model)
+    #num_Classes = 3
```
Comment (danielmao2019) [inline models/cnn/AlexNet/test_AlexNet.py]: No need to run too much test cases. These test cases are all equivalent. If one fails then all fails. If one passes then you know that all other can pass.

**Review by danielmao2019 — CHANGES_REQUESTED** [review]

File: models/cnn/AlexNet/test_AlexNet.py [hunk 4]
```diff
@@ -1,21 +1,36 @@
...
+def test_forward_pass(input_shape):
     logging.info(f"[RUNNING] Test case 'forward_pass' started.")
+    model = AlexNet(num_classes=10)
+    model = model.build(input_shape=input_shape)
     input = tf.zeros(shape=(1,)+model.layers[0].input_shape[0][1:])
+    logging.debug(f"{input.shape=}")
     output = model(input)
     assert len(output.shape) == 2, f"{output.shape=}"
     assert output.shape[0] == input.shape[0], f"{output.shape=}"
     logging.info("[PASSED] Test case 'forward_pass' passed.")
```
Comment (danielmao2019) [inline models/cnn/AlexNet/test_AlexNet.py:23]: no need the two lines of logging.info. please remove.

File: models/cnn/AlexNet/test_AlexNet.py [hunk 5]
```diff
@@ -1,21 +1,36 @@
...
 
-
-def test_overfit(model):
+@pytest.mark.dependency(depends=['test_forward_pass'])
+@pytest.mark.slow
+@pytest.mark.parametrize("input_shape", [
+    (128, 128, 3),
+])
+def test_overfit(input_shape):
     logging.info(f"[RUNNING] Test case 'overfit' started.")
```
Comment (danielmao2019) [inline models/cnn/AlexNet/test_AlexNet.py:31]: no need this info.

File: models/cnn/AlexNet/test_AlexNet.py [hunk 6]
```diff
@@ -33,3 +48,4 @@ def test_overfit(model):
     model = model.build(input_shape=(128, 128, 3))
     test_forward_pass(model)
```
Comment (danielmao2019) [inline models/cnn/AlexNet/test_AlexNet.py:64]: This is not correct. Please follow SegNet and ENet testing code.

**Review by danielmao2019 — CHANGES_REQUESTED** [review]

File: models/cnn/AlexNet/test_AlexNet.py [hunk 7]
```diff
@@ -1,21 +1,37 @@
...
+@pytest.mark.parametrize("input_shape", [
+    (128, 128, 3),
+])
+def test_overfit(input_shape):
     logging.info(f"[RUNNING] Test case 'overfit' started.")
+    model = AlexNet(num_classes=10)
+    model = model.build(input_shape=input_shape)
+    loss = tf.keras.losses.SparseCategoricalCrossentropy()
     model.compile(
         loss=tf.keras.losses.SparseCategoricalCrossentropy(),
```
Comment (danielmao2019) [inline models/cnn/AlexNet/test_AlexNet.py:36]: you don't need to repeat `tf.keras.losses.SparseCategoricalCrossentropy()`. just pass the variable `loss` above.

**Review by danielmao2019 — CHANGES_REQUESTED** [review]
